kernel/arch/aarch64/boot/exc_dispatch.c

```
@@ -0,0 +1,35 @@
+
+/* Exception slot indices for IRQ at each EL */
+#define SLOT_EL1_IRQ  5      /* Current EL SP_EL1, IRQ */
+#define SLOT_EL0_IRQ  9      /* Lower EL AArch64, IRQ  */
+#define MAX_HANDLERS  1024u  /* GICv2 INTID space, including spurious 1023 */
+
+static void (*s_handlers[MAX_HANDLERS])(void);
+
+void aarch64_exc_register_handler(int slot, void (*handler)(void)) {
+    /* Registration is a boot-time operation: callers must install handlers
+     * before unmasking IRQs with DAIFCLR so dispatch can read lock-free. */
+    if (slot >= 0 && (uint32_t)slot < MAX_HANDLERS)
```

Clang (14.0.6)

[warning] statement should be inside braces

(readability-braces-around-statements)